### ISSUE TITLE
Set correct application type under Linux

### ISSUE THREAD

**Issue by Sh1d0w** [brave/browser-laptop#2839]
**Did you search for similar issues before submitting this one?**

Yes

**Describe the issue you encountered:**

Currently the brave.desktop launcher which is generated, is not well formated so Gnome desktop for example could not recognize Brave as a web browser, thus it does not appear in default applications list. When you fix this is will allow the users to set Brave as default browser from the Gnome preferences for default applications

Currently this is what we have:

    cat /usr/share/applications/brave.desktop        

    [Desktop Entry]
    Name=brave
    Comment=Brave laptop and desktop browser
    GenericName=brave
    Exec=brave %U
    Icon=brave
    Type=Application
    StartupNotify=true
    Categories=Network;
    MimeType=text/plain;

**Expected behavior:**

This is what it should look like as a browser application:

    [Desktop Entry]
    Name=brave
    Comment=Brave laptop and desktop browser
    GenericName=brave
    Exec=brave %U
    Icon=brave
    Type=Application
    StartupNotify=true
    Categories=Network;WebBrowser;
    MimeType=text/html;text/xml;application/xhtml_xml;image/webp;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;

- Platform (Win7, 8, 10? macOS? Linux distro?):

Linux, Fedora 24 Gnome, Ubuntu 16.04

- Brave Version:
0.11.1
- Steps to reproduce:
    1. In gnome shell search for Details
    2. Click on Default Applications

- Screenshot if needed:

- Any related issues:
#1336

**Comment by willy-b** [comment]
@Sh1d0w, I just submitted a PR that sets the MimeType field to what you provided. (At least on Debian, as `electron-installer-redhat` doesn't yet allow the MimeType in the `.desktop` file to be configured.)

For comparison, the MimeType entry for Firefox's `.desktop` file looks like:
```
MimeType=text/html;text/xml;application/xhtml+xml;application/xml;application/rss+xml;application/rdf+xml;image/gif;image/jpeg;image/png;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;x-scheme-handler/chrome;video/webm;application/x-xpinstall;
```
(from `firefox-mozilla-build.desktop` in `firefox-mozilla-build_47.0.1-0ubuntu1_i386.deb`)

The WebP format looks awesome. Should we also add image/gif and image/png? Or should we leave that to other apps?

**Comment by Sh1d0w** [comment]
@willy-b Thanks for your response.

I have just opened a PR to fix this in `electron-installer-redhat` https://github.com/unindented/electron-installer-redhat/pull/29

When it gets merged you will be able to set this for the RPM builds as well.

No, I don't think you will need another image MIME-types listed here, as Brave is a web browser application in general. The proposed MimeTypes that I have suggested are taken from `google-chrome.desktop`
